spec/lib/diaspora/fetcher/public_spec.rb

```diff
@@ -100,7 +134,7 @@
       before do
         Timecop.freeze
         @now = DateTime.now.utc
-        @data = JSON.parse(@fixture).select {|item| item["post_type"] == "StatusMessage" }
+        @data = fixture_data.select {|item| item["post_type"] == "StatusMessage" }
```

Comment:
This shouldn't be filtered to only `StatusMessage` anymore, as we also fetch/import reshares now.

So both the `it "applies the date from JSON to the record"` and the `it "applies now to interacted_at on the record"` tests below can directly access `fixture_data` instead of `@data`, and instead of `StatusMessage.find_by` use `Post.find_by`.

The `it "copied the text correctly"` still only applies for `StatusMessage`, as reshares don't have a `text`, but then you can just replace `@data` with `fixture_data.select {|item| item["post_type"] == "StatusMessage" }` only in that test locally (and maybe rename it to `it "copied the text of status messages correctly"`.

And then you don't need `@data` anymore in the `before` block here.